includes/connect/class-wc-stripe-connect-api.php

```diff
@@ -0,0 +1,244 @@
+				$business_data['city']           = '';
+				$business_data['state']          = $base_location['state'];
+				$business_data['zip']            = '';
+			}
+
+			$request = array(
+				'returnUrl'    => $return_url,
+				'businessData' => $business_data,
+			);
+
+			return $this->request( 'POST', '/stripe/oauth-init', $request );
+		}
+
+		/**
+		 * Send request to Connect Server for Stripe keys
+		 *
+		 * @param  string $code OAuth server code.
+		 *
+		 * @return array
+		 */
+		public function get_stripe_oauth_keys( $code ) {
+
+			$request = array( 'code' => $code );
+
+			return $this->request( 'POST', '/stripe/oauth-keys', $request );
+		}
+
+		/**
+		 * General OAuth request method.
+		 *
+		 * @param string $method request method.
+		 * @param string $path   path for request.
+		 * @param array  $body   request body.
+		 *
+		 * @return array|WP_Error
+		 */
+		protected function request( $method, $path, $body = array() ) {
+
+			if ( ! is_array( $body ) ) {
+				return new WP_Error(
+					'request_body_should_be_array',
+					__( 'Unable to send request to WooCommerce Connect server. Body must be an array.', 'woocommerce-gateway-stripe' )
+				);
+			}
+
+			$url = trailingslashit( WOOCOMMERCE_CONNECT_SERVER_URL );
+			$url = apply_filters( 'wc_connect_server_url', $url );
+			$url = trailingslashit( $url ) . ltrim( $path, '/' );
+
+			// Add useful system information to requests that contain bodies.
+			if ( in_array( $method, array( 'POST', 'PUT' ), true ) ) {
+				$body = $this->request_body( $body );
+				$body = wp_json_encode( apply_filters( 'wc_connect_api_client_body', $body ) );
+
+				if ( ! $body ) {
+					return new WP_Error(
+						'unable_to_json_encode_body',
+						__( 'Unable to encode body for request to WooCommerce Connect server.', 'woocommerce-gateway-stripe' )
+					);
+				}
+			}
+
+			$headers = $this->request_headers();
+			if ( is_wp_error( $headers ) ) {
+				return $headers;
+			}
+
+			$http_timeout = 60; // 1 minute
+			if ( function_exists( 'wc_set_time_limit' ) ) {
+				wc_set_time_limit( $http_timeout + 10 );
+			}
+			$args = array(
+				'headers'     => $headers,
+				'method'      => $method,
+				'body'        => $body,
+				'redirection' => 0,
+				'compress'    => true,
+				'timeout'     => $http_timeout,
+			);
+
+			$args          = apply_filters( 'wc_connect_request_args', $args );
+			$response      = wp_remote_request( $url, $args );
+			$response_code = wp_remote_retrieve_response_code( $response );
+			$content_type  = wp_remote_retrieve_header( $response, 'content-type' );
+
+			if ( false === strpos( $content_type, 'application/json' ) ) {
+				if ( 200 !== $response_code ) {
+					return new WP_Error(
+						'wcc_server_error',
+						sprintf(
+							// Translators: HTTP error code.
+							__( 'Error: The WooCommerce Connect server returned HTTP code: %d', 'woocommerce-gateway-stripe' ),
+							$response_code
+						)
+					);
+				}
+				return $response;
```

Comment:
What happens if we exit this way? Seems bad.

Reply:
Good catch! I'm not sure why this was returned originally, but this should be an error. I've addressed this in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1267/commits/c18c5527253345618ac522a6a3a34ddb6d2ceb3d